lib/pre-build.js

```diff
@@ -113,10 +103,10 @@ function setupRequest(url, fileName) {
 preBuild.download = async function download(target) {
   const fileName = getPackageFileName(target)
   const url = DOWNLOAD_HOST + REMOTE_PATH + fileName
-  const { client, options } = setupRequest(url, fileName)
+  const { client, options } = setupRequest()
 
   return new Promise((resolve, reject) => {
-    client.get(options, function handleResponse(res) {
+    client.get(url, options, function handleResponse(res) {
```

Comment:
this is to appease the Node 20 gods. It seems like making an agent in options is ok in pre 20 but not in 20 so just split out url and http(s) options